sway-core/src/error.rs

```diff
@@ -538,6 +538,10 @@ pub enum CompileError {
         variable_name: Ident,
         span: Span,
     },
+    #[error(
+        "This parameter was declared as mutable, which is not supported yet, did you mean to use ref mut?"
+    )]
```

Comment:
I would argue we'll never support mutable parameters, the `ref` bit is pretty much documentation.  But that's not a deal breaker.